### DISCUSSION TITLE
be able to set goals for day ?

### DISCUSSION THREAD

**Discussion by spcbfr** [monkeytypegame/monkeytype#1659] [Ideas] [answered]
It would be nice if the user could set a daily / weekly goal practice time goal and it would congratulate you once the goal is achieved, maybe even have a % in the menu to show how much you completed

**Comment by Miodec** [comment, marked as answer]
I guess it would be nice but I'm struggling to find a good placement in the UI for this. Also even if - this is probably very far in the future.

**Reply by unrealapex** [reply]
@Miodec how about these locations?
![here](https://user-images.githubusercontent.com/79121360/126729314-8f58ec56-532f-4e82-baed-90ef01170d69.png)
![here2](https://user-images.githubusercontent.com/79121360/126729315-a2aa61d5-e5d5-4a50-a08b-35f9e0008fde.png)

Here are some ideas of what it could look like:
##### Daily goal
![goal](https://user-images.githubusercontent.com/79121360/126729428-716aab45-d03c-40e3-b3bb-5eb45e9598c1.png)
##### Modifying the daily goal (upon clicking the goal, a dropdown appears)
![goal2](https://user-images.githubusercontent.com/79121360/126729425-ded2a06c-23ff-411f-a110-3f5ee8d8d89e.png)

**Reply by spcbfr** [reply]
looks good !

**Reply by Otard95** [reply]
Something like this could also work, not sure about the colors but that can change

![monkeytype-goals](https://user-images.githubusercontent.com/7838337/126784750-06e075e6-90ea-44a2-994c-a6e13d2329d9.gif)

EDIT: Ohh that gif was glitchy xD

**Reply by spcbfr** [reply]
I think it would be better if it would be visible without the user clicking

**Reply by unrealapex** [reply]
I agree as less clicks is always better in terms of ui ux design, but the concept is good. Putting it next to the session time seems good.

**Reply by Otard95** [reply]
> I think it would be better if it would be visible without the user clicking

True, it could be static but expanded to the right instead. However that might make the page feel lopsided. 

Another option is a scroll interaction. Scroll down makes the stats smaller and some fade and move combo reveals the goals. 

Main thing was just to keep it clean and clutter free

**Comment by elburro1887** [comment]
I would also be interested in this, a lot of other typing apps have this feature and it helps the user stay focused and eager to study.
